new-log-viewer/src/App.tsx

```diff
@@ -11,7 +12,9 @@ const App = () => {
     return (
         <>
             <StateContextProvider>
-                <Layout/>
+                <UrlContextProvider>
+                    <Layout/>
+                </UrlContextProvider>
             </StateContextProvider>
```

Comment:
Supposedly the <StateContextProvider/> would subscribe to changes in the URL, right? To allow <StateContextProvider/> to use the context, would it make more sense to have UrlContextProvider as the root node?

Reply:
You're absolutely right. The state depends on the specific URL.